src/Domain/Structure/ElementSearchTree.hpp

```diff
@@ -0,0 +1,100 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/index/rtree.hpp>
+#include <cstddef>
+#include <map>
+#include <utility>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Structure/ElementId.hpp"
+
+namespace bg = boost::geometry;
+
+namespace boost::geometry::traits {
+// Make Tensor compatible with Boost.Geometry
```

Comment:
Can you add some documentation of why you need to do this for Tensor? ElementID seems "obvious"